appserver/admingui/common/src/main/java/org/glassfish/admingui/common/handlers/ApplicationHandlers.java

```diff
@@ -80,6 +84,17 @@
     private static final String NAME_CONFIG_NAME = "configName";
     private static final String NAME_RESULT = "result";
 
+    @Handler(id = "py.getContextRoot",
+            input = {
+                @HandlerInput(name = "appName", type = String.class, required = true)},
+            output = {
+                @HandlerOutput(name = NAME_RESULT, type = String.class)})
+    public static void getContextRoot(HandlerContext handlerCtx) {
+        Applications applications = Globals.getDefaultBaseServiceLocator().getService(Applications.class);
+        String appName = (String) handlerCtx.getInputValue("appName");
+        Application app = applications.getApplication(appName);
+        handlerCtx.setOutputValue(NAME_RESULT, app.getContextRoot());
```

Comment:
@Pandrex247 @MattGill98 brought up the question of including the virtual server context root here to create working URLs in case the virtual server uses a root path. WDYT?

Reply:
Looking at `org.glassfish.admingui.common.handlers.ApplicationHandlers.getTargetURLList(String, String)` I'd think that line 772 takes care of it. So while this resolves the "plain" context root of the application the URLs are still resolved using the `getTargetURLList` handler which calls the referenced code which takes VS into account.

Reply:
This is still a relevant topic, I just changed the line for other reasons.